### PR TITLE
Explain masked arrays in sigma_clip

### LINKED ISSUE #3858: Better explain masked arrays in sigma_clip
For many users, sigma_clip may be where they encounter masked arrays for the first time. It would be nice to improve the docstring for sigma_clip:

http://docs.astropy.org/en/stable/api/astropy.stats.sigma_clip.html#astropy.stats.sigma_clip

to include a few lines on what a masked array is (including a printed view), just before we mention that Numpy ufuncs often understand them. Alternatively, we could link to http://docs.scipy.org/doc/numpy/reference/maskedarray.html but that is a bit extensive.

### PR DESCRIPTION
**Pull request by Telomelonia**
### Description

Added explanation for masked array in index.rst for docs

resolves #3858 

### TODOs
- [x] Explanation about masked array in the documentation.
- [x] Fix the doctest failure.
- [x] Squash all commits into one commit.
- [x] Rebase against upstream main.
### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by Telomelonia** [comment]
There is a short description of the sigma clipping in [robust statistical estimator](https://github.com/astropy/astropy/blob/main/docs/stats/robust.rst#sigma-clipping). I was thinking to refer the "masked array" to the changes I made.

File: docs/stats/index.rst [hunk 1]
```diff
@@ -28,9 +28,47 @@ they can be accessed by importing them::
...
+
+    >>> outliers = clipped.data[clipped.mask]
+    >>> outliers
+    array([99])
+    >>> valid = clipped.data[~clipped.mask]
+    >>> valid
+    array([1, 0, 0, 1, 0, 0, 1, 0])
+
+For more information on masked arrays, including see the
+`Numpy documentation<https://numpy.org/doc/stable/reference/maskedarray.generic.html>`_.
```
Comment (pllim) [inline docs/stats/index.rst:71]: I think we can use intersphinx here.
```suggestion
:ref`numpy:maskedarray.generic`.
```

**Comment by pllim** [comment]
Also, thanks for trying to resolve a very old issue!

File: docs/stats/index.rst (hunk 1 above)
Comment (Telomelonia) [reply docs/stats/index.rst:71]: Do this requires any intersphinx mapping in conf.py

**Comment by Telomelonia** [comment]
> Also, thanks for trying to resolve a very old issue!

Well, I am new to this repo and wanted to contribute... so I started to resolve old doc issues to get an idea about the repo. Thanks for assisting in this PR ✌️.

File: docs/stats/index.rst (hunk 1 above)
Comment (pllim) [reply docs/stats/index.rst:71]: We already have it. 😸 

https://github.com/astropy/sphinx-astropy/blob/2d2b019c99e02ef38feeb16d28a19610abe499d9/sphinx_astropy/conf/v1.py#L60

**Review by larrybradley — CHANGES_REQUESTED** [review]
Thanks, @Telomelonia. I had just a couple minor comments.

**Comment by pllim** [comment]
1. Fix the doctest failure.
2. Squash all commits into one commit.
3. Rebase against upstream main.

**Comment by Telomelonia** [comment]
@pllim 
Please check the PR ... all the tasks are done and all CIs have passed. ✌️

**Comment by pllim** [comment]
@Telomelonia , I still see 9 commits, including merge commits that are undesirable. Please rebase (and resolve any conflict that arises though I don't think there are any):

```
git fetch upstream main
git rebase upstream/main
```

Then use `git rebase -i HEAD~N` to squash all your remaining commits into a single commit.

**Review by larrybradley — COMMENTED** [review]
Thanks, @Telomelonia.  Please squash to 1 commit and rebase on the current `main`.

**Review by larrybradley — APPROVED** [review]

**Review by WilliamJamieson — CHANGES_REQUESTED** [review]
Please remove the merge commit 09c7ede from your commit history. We request that you use rebases instead of merges.

**Comment by Telomelonia** [comment]
@WilliamJamieson 
On it ... Actually I made a mistake of force merging main

**Pull request closed without merging**